lib/Common/Exceptions/ExceptionCheck.cpp

```diff
@@ -58,7 +58,8 @@ void ExceptionCheck::SetHandledExceptionType(ExceptionType e)
 #if DBG
     if(!(e == ExceptionType_None ||
          e == ExceptionType_DisableCheck ||
-         !JsUtil::ExternalApi::IsScriptActiveOnCurrentThreadContext() ||
+         (e & ExceptionType_OutOfMemory) == ExceptionType_OutOfMemory ||  // InitializeModuleRecord handles OOM during dynamic import
```

Comment:
>InitializeModuleRecord [](start = 77, length = 22)

Why add this?  The point here is that if you are in script, you should be throwing the JS exception object instead of OutOfMemory.